From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yive <6853318+Yive@users.noreply.github.com>
Date: Sun, 18 Feb 2024 02:33:47 -0800
Subject: [PATCH] Configuration for fluid ticks

Controls how often fluids are ticked, such as for spreading to new locations. A value below 1 will stop all fluid ticks

diff --git a/src/main/java/dev/yive/pluto/PlutoConfig.java b/src/main/java/dev/yive/pluto/PlutoConfig.java
index 3ce4faa60abcb2f116cd0f6f3fd560a1480f2b28..f867010621a22ca310b35d78704ac0feceacf706 100644
--- a/src/main/java/dev/yive/pluto/PlutoConfig.java
+++ b/src/main/java/dev/yive/pluto/PlutoConfig.java
@@ -352,5 +352,10 @@ public class PlutoConfig {
         private void blockTicking() {
             blockTickingModulo = getInt("blocks.global.block-ticking-modulo", blockTickingModulo);
         }
+
+        public int fluidTickingModulo = 1;
+        private void fluidTicking() {
+            fluidTickingModulo = getInt("blocks.fluids.fluid-ticking-modulo", fluidTickingModulo);
+        }
     }
 }
diff --git a/src/main/java/net/minecraft/server/level/ServerLevel.java b/src/main/java/net/minecraft/server/level/ServerLevel.java
index 98fd10e930f6dbf4e3ccb6b12eafcf1652606bac..d8ddf2ffbedb6479c3ffa6b82e8137219301f708 100644
--- a/src/main/java/net/minecraft/server/level/ServerLevel.java
+++ b/src/main/java/net/minecraft/server/level/ServerLevel.java
@@ -759,6 +759,7 @@ public class ServerLevel extends Level implements ServerEntityGetter, WorldGenLe
             if (plutoConfig().blockTickingModulo > 0 && (plutoConfig().blockTickingModulo == 1 || MinecraftServer.currentTick % plutoConfig().blockTickingModulo == 0)) // Pluto - Configuration for block ticks
             this.blockTicks.tick(j, paperConfig().environment.maxBlockTicks, this::tickBlock); // Paper - configurable max block ticks
             gameprofilerfiller.popPush("fluidTicks");
+            if (plutoConfig().fluidTickingModulo > 0 && (plutoConfig().fluidTickingModulo == 1 || MinecraftServer.currentTick % plutoConfig().fluidTickingModulo == 0)) // Pluto - Configuration for fluid ticks
             this.fluidTicks.tick(j, paperConfig().environment.maxFluidTicks, this::tickFluid); // Paper - configurable max fluid ticks
             gameprofilerfiller.pop();
         }
